includes/Checker/Preparations/Force_Single_Plugin_Preparation.php

```diff
@@ -74,39 +75,67 @@ public function prepare() {
 	}
 
 	/**
-	 * Filter active plugins.
+	 * Filters active plugins to only include required ones.
+	 *
+	 * This means:
+	 *
+	 * * The plugin being tested
+	 * * All dependencies of the plugin being tested
+	 * * Plugin Check itself
+	 * * All plugins depending on Plugin Check (they could be adding new checks)
 	 *
 	 * @since 1.0.0
 	 *
-	 * @param array $active_plugins List of active plugins.
-	 * @return array List of active plugins.
+	 * @param mixed $active_plugins List of active plugins.
+	 * @return mixed List of active plugins.
 	 */
 	public function filter_active_plugins( $active_plugins ) {
-		if ( is_array( $active_plugins ) && in_array( $this->plugin_basename, $active_plugins, true ) ) {
+		if ( ! is_array( $active_plugins ) ) {
+			return $active_plugins;
+		}
 
-			if ( defined( 'WP_PLUGIN_CHECK_MAIN_FILE' ) ) {
-				$plugin_check_file = WP_PLUGIN_CHECK_MAIN_FILE;
-			} else {
-				$plugins_dir       = defined( 'WP_PLUGIN_DIR' ) ? WP_PLUGIN_DIR : WP_CONTENT_DIR . '/plugins';
-				$plugin_check_file = $plugins_dir . '/plugin-check/plugin.php';
-			}
+		// The plugin being tested isn't actually active yet.
+		if ( ! in_array( $this->plugin_basename, $active_plugins, true ) ) {
+			return $active_plugins;
+		}
 
-			$plugin_base_file = plugin_basename( $plugin_check_file );
+		if ( defined( 'WP_PLUGIN_CHECK_MAIN_FILE' ) ) {
+			$plugin_check_file = WP_PLUGIN_CHECK_MAIN_FILE;
+		} else {
+			$plugins_dir       = defined( 'WP_PLUGIN_DIR' ) ? WP_PLUGIN_DIR : WP_CONTENT_DIR . '/plugins';
+			$plugin_check_file = $plugins_dir . '/plugin-check/plugin.php';
+		}
 
-			// If the plugin-check is the only available plugin then return that one only.
-			if ( $this->plugin_basename === $plugin_base_file ) {
+		$plugin_check_basename = plugin_basename( $plugin_check_file );
 
-				return array(
-					$plugin_base_file,
-				);
-			}
+		$new_active_plugins = array(
+			$this->plugin_basename, // Plugin to test.
+			$plugin_check_basename, // Plugin Check itself.
+		);
 
-			return array(
-				$this->plugin_basename,
-				$plugin_base_file,
+		// Plugin dependencies support was added in WordPress 6.5.
+		if ( class_exists( 'WP_Plugin_Dependencies' ) ) {
+			WP_Plugin_Dependencies::initialize();
+
+			$new_active_plugins = array_merge(
+				$new_active_plugins,
+				WP_Plugin_Dependencies::get_dependencies( $this->plugin_basename )
+			);
+
+			$new_active_plugins = array_merge(
+				$new_active_plugins,
+				// Include any dependents of Plugin Check, but only if they were already active.
+				array_filter(
+					WP_Plugin_Dependencies::get_dependents( dirname( $plugin_check_basename ) ),
+					static function ( $dependent ) use ( $active_plugins, $new_active_plugins ) {
+						return in_array( $dependent, $active_plugins, true ) ||
+								in_array( $dependent, $new_active_plugins, true );
```

Comment:
Why do we need the second condition here? That seems pointless to me, since we're checking for existence in `$new_active_plugins` which is the same array we want to potentially add these plugins to.
```suggestion
						return in_array( $dependent, $active_plugins, true );
```

Reply:
The dependent could be the plugin being checked, so it would already be in `$new_active_plugins`.

But I'm also doing a `array_unique()` below, so it shouldn't matter.

Reply:
Yeah, that's what I was thinking. I was initially hesitant about using `array_unique()` (because before this was just one single string comparison instead), but since you use it, no need to prevent duplicates here either.